# Departamento de marketing

- Dados: https://www.kaggle.com/arjunbhasin2013/ccdata



## Importação das bibliotecas


In [1]:
import pandas as pd
import numpy as np #Funções matemáticas
import seaborn as sns #Gerar gráficos
import matplotlib.pyplot as plt #Gerar gráficos
from sklearn.preprocessing import StandardScaler #Colocar os dados no mesmo padarão (mesma escala)
from sklearn.cluster import KMeans #Realizar a segmentação do mercado / Agrupamento dos clientes
from sklearn.decomposition import PCA #Reduzir a dimensionalidade dos dados

In [2]:
creditcard_df = pd.read_csv('/content/marketing_data.csv')
print(f'{creditcard_df.shape[1]} características que os {creditcard_df.shape[0]} clientes terão')

18 características que os 8950 clientes terão


In [3]:
creditcard_df.head()

CUST_ID      BALANCE  ...  PRC_FULL_PAYMENT  TENURE
0  C10001    40.900749  ...          0.000000      12
1  C10002  3202.467416  ...          0.222222      12
2  C10003  2495.148862  ...          0.000000      12
3  C10004  1666.670542  ...          0.000000      12
4  C10005   817.714335  ...          0.000000      12

[5 rows x 18 columns]

In [4]:
creditcard_df.describe()

BALANCE  BALANCE_FREQUENCY  ...  PRC_FULL_PAYMENT       TENURE
count   8950.000000        8950.000000  ...       8950.000000  8950.000000
mean    1564.474828           0.877271  ...          0.153715    11.517318
std     2081.531879           0.236904  ...          0.292499     1.338331
min        0.000000           0.000000  ...          0.000000     6.000000
25%      128.281915           0.888889  ...          0.000000    12.000000
50%      873.385231           1.000000  ...          0.000000    12.000000
75%     2054.140036           1.000000  ...          0.142857    12.000000
max    19043.138560           1.000000  ...          1.000000    12.000000

[8 rows x 17 columns]

### **Balance**
Os clientes mantém na conta corrente em média 1564.47$ para fazer compras no cartão de débito

### **Balance_Frequency**
Em média a frequencia de atualização do saldo da contaa corrente é de 87,7% 

### **Purchases**
Valor médio das compras é de 1000$

### **Comparação entre ONEOFF_PURCHASES (a vista) e INSTALLMENTS_PURCHASES (parcelado)**
A vista: em média 592$  

Parcelado: em média 411$  

*Pode ser um grupo de pessoas*

### **Cash_Advance**
Em média, as pessoas sacam mais dinheiro do limite do cartão de crédito do que fazer compras parceladas e a vistas  

Pode ser ruim pros clientes mas é bom pro banco pq os juros são altos  

*Pode ser um grupo de pessoas*

### **Comparação entre ONEOFF_PURCHASES_FREQUENCY e PURCHASES_INSTALLMENTS_FREQUENCY**
As pessoas realizam mais compras parcelado (0.364437) do que a vista (0.202458)

### **PRC_FULL_PAYMENT**
15% dos clientes pagam o valor completo da fatura



## Visualização e exploração dos dados

## Definição do número de clusters usando o Elbow Method

- Mais detalhes 
  - https://en.wikipedia.org/wiki/Elbow_method_(clustering)
  - https://www.geeksforgeeks.org/elbow-method-for-optimal-value-of-k-in-kmeans/

## Agrupamento com k-means

## Aplicação de autoencoders